src/routes/mission/[...mission]/+page.svelte

```diff
@@ -72,6 +81,40 @@
 		};
 	}
 
+	let demilClient = new DeMiL.DeMiLClient(8095);
+	async function startMission() {
+		demilHelpState = undefined;
+		try {
+			const missionResult = await demilClient.startMissionByName(mission.name, steamID);
+			demilVersion = !missionResult.IsVersionInRange ? missionResult.Version ?? '<2.1.0' : undefined;
+			if (missionResult.hasOwnProperty('MissingModules')) {
+				missingModules = (missionResult as DeMiL.StartMissionMissingModulesResult).MissingModules.map(mod =>
+					getModule(mod, modules)
+				);
+				demilHelpState = 'MissingModules';
+			} else if (missionResult.hasOwnProperty('MissionModulesCount')) {
+				let moduleCountError = missionResult as DeMiL.StartMissionTooManyModulesResult;
+				demilErrorMessage = `Failed to start mission. A bomb that can support more modules is required. Current bombs only support up to ${moduleCountError.MaximumSupportedModulesCount} modules, and the mission has ${moduleCountError.MissionModulesCount} modules.`;
+				demilHelpState = 'Error';
+			} else if (missionResult.hasOwnProperty('MissionBombsCount')) {
+				let bombCountError = missionResult as DeMiL.StartMissionTooManyBombsResult;
+				demilErrorMessage = `A room that can support more bombs is required. Current rooms only support up to ${bombCountError.MaximumSupportedBombsCount} bombs, and the mission has ${bombCountError.MissionBombsCount} bombs.`;
+				demilHelpState = 'Error';
+			} else {
+				toast.success(`Started mission ${missionResult.MissionID}`);
```

Comment:
I think literally all of this from line 84 to 117 can be in the component instead of in the mission page code and you just pass it the mission object.